File: Backend old/src/main/java/com/youcode/MonSupplier/models/Entities/Product.java
package com.youcode.MonSupplier.models.Entities;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@Entity
@Table(name = "product")
public class Product {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String name;
    private float prix;
    private int less_quantity;
    private int quantity;
    private String description;

    @ManyToOne
    private Supplier supplier;

    @ManyToOne
    private Category category;
}

File: frontend/src/app/models/product.ts
import { Supplier } from "./supplier";
import { Category } from "./category";
export interface Product {
image: any;
    id: number;
    name: string;
    prix: number;
    less_quantity: number;
    quantity: number;
    description: string;
    supplier: Supplier;
    category: Category;
}

